app/routes.py

```diff
@@ -1 +1,161 @@
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = [task.task_to_dict() for task in tasks]
+    return jsonify(tasks_response)
+
+@task_list_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_model_task(Task, task_id)
+    response_body = {
+        "task": task.task_to_dict()
+    }
+    return jsonify(response_body)
+
+@task_list_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model_task(Task, task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    response_body = {
+        "task": task.task_to_dict()
+    }
+    return response_body
+
+@task_list_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model_task(Task, task_id)
+    response_body = {
+        "details": f'Task {task.task_id} \"{task.title}\" successfully deleted'
+    }
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return response_body
+
+@task_list_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    request_body = request.get_json()
+
+    try:
+        task = Task.query.get(task_id)
+        task.completed_at = now()
+    except:
+        return abort(make_response({"message": f"{task_id} not found"}, 404))
+
+    db.session.commit()
+
+    response_body = {
+        "task": task.task_to_dict()
+    }
+
+    SLACK_BOT_TOKEN = os.environ.get("SLACK_BOT_TOKEN")
+    my_headers = {
+        "Authorization": f"Bearer {SLACK_BOT_TOKEN}"
+    }
+
+    data_payload = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.title}"
+    }
+
+    requests.post("https://slack.com/api/chat.postMessage",
+                    headers=my_headers,
+                    data=data_payload)
+    
+    return jsonify(response_body)
+
+
+    # ~~Below is the the code I would have used given Slack documentation for calling their API~~
+
+    # client = WebClient(token=os.environ.get("SLACK_BOT_TOKEN"))
+    # logger = logging.getLogger(__name__)
+    # channel_id = "task-notifications"
+
+    # try:
+    #     # Call the chat.postMessage method using the WebClient
+    #     result = client.chat_postMessage(
+    #         channel=channel_id, 
+    #         text=f"Someone just completed the task {task.title}."
+    #     )
+    #     logger.info(result)
+
+    # except SlackApiError as e:
+    #     logger.error(f"Error posting message: {e}")
+
+    
+
+@task_list_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    request_body = request.get_json()
+    try:
+        task=Task.query.get(task_id)
+        task.completed_at = None
+    except:
+        response_body = abort(make_response({"message": f"{task_id} not found"}, 404))
+        return response_body
```

Comment:
I believe `abort` has an implicit return, so we can combine these two lines together:
```python
        abort(make_response({"message": f"{task_id} not found"}, 404))
```